src/libertem/udf/stddev.py

```diff
@@ -130,20 +214,31 @@ def merge(self, dest, src):
             Partial results that contains sum of variances, sum of frames, and the
             number of frames used over current iteration of partition
         """
-        N0 = _validate_n(dest['num_frame'][0])
-        N1 = _validate_n(src['num_frame'][0])
-        p0 = VariancePart(var=dest['var'][:],
-                        sum_im=dest['sum_frame'][:],
-                        N=N0)
-        p1 = VariancePart(var=src['var'][:],
-                        sum_im=src['sum_frame'][:],
-                        N=N1)
-        compute_merge = merge(p0, p1)
-
-        dest['var'][:] = compute_merge.var
-        dest['sum_frame'][:] = compute_merge.sum_im
+        dest_N = _validate_n(dest['num_frame'][0])
+        src_N = _validate_n(src['num_frame'][0])
+
+        oldshape = dest['sum_frame'][:].shape
+
+        dest['sum_frame'][:].shape = (-1,)
+        dest['var'][:].shape = (-1, )
+        src['sum_frame'][:].shape = (-1,)
+        src['var'][:].shape = (-1, )
```

Comment:
Today I learned that `flatten()` creates a copy while `reshape((-1, ))` creates a view if possible. It works with `reshape()` instead of assigning to `shape`.

Reply:
Yes :+1: :+1: I regularly reshape flat arrays before writing to them, works really well! AFAIK it will just assign different strides and dimensions on the ndarray object.